llvm/lib/IR/RepoTicket.cpp

```diff
@@ -139,12 +139,18 @@ static bool updateDigestUseDependenciesAndContributions(
   GOHash.update('T');
   auto State = AccumulateGODigest(GO, GOHash, GOIMap);
   Changed = Changed || State.Changed;
-  // Update the GO's hash using its contributions and dependencies.
-  auto &GOContributions = State.Contributions;
-  auto &GODependencies = State.Dependencies;
-  for (const GlobalObject *const G : llvm::concat<const GlobalObject *const>(
-           make_range(GOContributions.begin(), GOContributions.end()),
-           make_range(GODependencies.begin(), GODependencies.end()))) {
+  // Add the GO's contributions to its hash. Contributions can't form loops
+  // since 1) only global variables have contributions and 2) all contributions
+  // are functions.
+  for (const GlobalObject *const G : State.Contributions) {
+    assert(isa<GlobalVariable>(GO) &&
+           "Only global variables can have contributions!");
+    assert(isa<Function>(G) && "All contributions are functions!");
+    Changed = true;
```

Comment:
So `Changed ` is set to true if there are any contributions. Would this be clearer as:
`Changed = Changed || !State.Contributions.empty();`

before the loop to make it clear that it's loop invariant.

(I'm not sure which would be clearer but might be worth a try).

Reply:
Fixed, thanks.